game/scripts/vscripts/abilities/oaa_zuus_cloud.lua

```diff
@@ -0,0 +1,155 @@
+
+    ParticleManager:SetParticleControlEnt(self.cloud_particle, 2, self:GetParent(), PATTACH_POINT_FOLLOW, "attach_hitloc", self:GetParent():GetAbsOrigin(), true)
+    self:StartIntervalThink( 0.1 )
+    EmitSoundOn("Hero_Zuus.Cloud.Cast", self:GetParent())
+  end
+end
+
+function modifier_zuus_cloud_oaa:OnDestroy()
+  if IsServer() then
+    ParticleManager:DestroyParticle(self.cloud_particle, false)
+    ParticleManager:ReleaseParticleIndex(self.cloud_particle)
+    self.cloud_particle = nil
+  end
+end
+
+function modifier_zuus_cloud_oaa:DeclareFunctions()
+	local funcs =
+	{
+		MODIFIER_PROPERTY_INCOMING_DAMAGE_PERCENTAGE
+	}
+	return funcs
+end
+
+function modifier_zuus_cloud_oaa:GetModifierIncomingDamage_Percentage( params )
+  local attacker = params.attacker;
+  -- percentage base applied to the income damage
+  -- -100 = 100% block
+  -- -100 + K/damage = K damage
+  if attacker:IsHero() and attacker:IsRangedAttacker() then
+    return -100 + math.ceil(1000*2/params.damage)/10
+  elseif attacker:IsHero() and not attacker:IsRangedAttacker() then
+    return -100 + math.ceil(1000*4/params.damage)/10
+  else
+    return -100 + math.ceil(1000/params.damage)/10
+  end
+end
+
+function modifier_zuus_cloud_oaa:OnIntervalThink()
+  if self.LastStrike == nil or GameRules:GetDOTATime(false, false) - self.LastStrike > self.Interval then
+    local caster = self:GetCaster()
+    local targets = FindUnitsInRadius(
+      caster:GetTeamNumber(),
+      self:GetParent():GetAbsOrigin(),
+      nil,
+      self.Radius,
+      DOTA_UNIT_TARGET_TEAM_ENEMY,
+      DOTA_UNIT_TARGET_HERO,
+      DOTA_UNIT_TARGET_FLAG_NONE,
+      FIND_CLOSEST,
+      false
+    )
+    if #targets == 0 then
+      targets = FindUnitsInRadius(
+        caster:GetTeamNumber(),
+        self:GetParent():GetAbsOrigin(),
+        nil,
+        self.Radius,
+        DOTA_UNIT_TARGET_TEAM_ENEMY,
+        DOTA_UNIT_TARGET_CREEP,
+        DOTA_UNIT_TARGET_FLAG_NONE,
+        FIND_CLOSEST,
+        false
+      )
+    end
+
+    if #targets > 0 then
+      self:CastLightningBolt(targets[1])
+      self.LastStrike = GameRules:GetDOTATime(false, false)
+    end
+  end
+end
+
+function modifier_zuus_cloud_oaa:CastLightningBolt(target)
+  local caster = self:GetCaster()
+  local parent = self:GetParent()
+  local lightning_bolt_ability = caster:FindAbilityByName( 'zuus_lightning_bolt' )
+  local sight_radius =  0
+  if GameRules:IsDaytime() then
+    sight_radius = lightning_bolt_ability:GetSpecialValueFor("sight_radius_day")
+  else
+    sight_radius = lightning_bolt_ability:GetSpecialValueFor("sight_radius_night")
+  end
+  local sight_duration = lightning_bolt_ability:GetSpecialValueFor("sight_duration")
+
+  if lightning_bolt_ability:GetLevel() > 0 then
+    AddFOWViewer(caster:GetTeam(), target:GetAbsOrigin(), sight_radius, sight_duration, false)
+
+    local talent = caster:FindAbilityByName("special_bonus_unique_zeus_3")
+
+    local ministun_duration = 0.1
```

Comment:
According to the wiki, the base ministun is 0.2 seconds. Don't know how they would know though.

Reply:
Interesting... this is true. If I inspect the modifier filter I get actually 1.803999..
but If I set it to 0.2 I still get this number.
odd...

Reply:
Were you testing on a Strength hero by any chance?

Reply:
True!